### DISCUSSION TITLE
跟diesel比，谁跟快

### DISCUSSION THREAD

**Discussion by Trojanking123** [rbatis/rbatis#283] [General]

**Comment by zhuxiujia** [comment]
> _No description provided._

*  假设 数据库和服务器都是单独部署的服务器。
理论上diesel 更‘’快‘’，这个快指的是响应时间快。因为diesel底层是c实现的驱动而且diesel是线程模型，而rbatis底层用来sqlx拷贝的纯rust Future模型实现的驱动。所以说对比起来的话，并不算特别公平。

*  你说的快如果是说并发量的话，假设并发量100w级，服务器4核心16GB内存，数据库核心和内存无限大。

rbatis 理论上基于Future模型的Rbatis会相对更高（不管是tokio，async_std核心线程数大概等于cpu数，也就是4条核心线程）future的0开销的概念

Diesel 却开启100w条线程的话，耗费极大的内存不说，cpu占用也会达到100%的空跑状态，而且在努力的线程切换，linux也极有可能内存不足强杀进程)
* 不过也可以压测试试，注意不要吧 数据库和服务部署在同一台机器上，我不清楚diesel的底层c 驱动是否会直接走ffi ，而不走tcp调用

**Comment by zhuxiujia** [comment]
> _No description provided._

而且线程模型应该更加适合游戏行业，游戏应该需要更快的响应延迟，而并发量可以增加服务器减少压力。
协程更加适合微服务，通用的行业，它更加关注陡增的并发量级吧

**Comment by Trojanking123** [comment]
有长期维护的 打算吗

**Comment by zhuxiujia** [comment]
长期维护

**Comment by avrilko-go** [comment]
厉害了

**Comment by weiznich** [comment]
@zhuxiujia Please provide a reproducible benchmark for your claim that rbatis is faster than diesel for a large number of connections. Otherwise this is only a claim without any substance.

**Comment by zhuxiujia** [comment]
> @zhuxiujia Please provide a reproducible benchmark for your claim that rbatis is faster than diesel for a large number of connections. Otherwise this is only a claim without any substance.

My original meaning is "in theory". Theoretically, the concurrency of future is larger than that of the thread model. So far, no stress testing has been carried out.
but, Because diesel installation and configuration are difficult for me，Pressure test examples will be provided as soon as possible

**Comment by sunli829** [comment]
大佬在rust微信群里面吗？

**Comment by zhuxiujia** [comment]
> 大佬在rust微信群里面吗？

先加微信 zxj347284221 拉进群

**Comment by pickfire** [comment]
@zhuxiujia Maybe try adding `rbatis` in https://github.com/diesel-rs/diesel/pull/2507 to see if its faster?

By the way, are you in rust 众 telegram group?

**Comment by zhuxiujia** [comment]
> @zhuxiujia Maybe try adding `rbatis` in [diesel-rs/diesel#2507](https://github.com/diesel-rs/diesel/pull/2507) to see if its faster?
> 
> By the way, are you in rust 众 telegram group?


The Rbatis depend on 'SQLX-core 0.4.0-beta.1',    There are currently no plans to replace the underlying implementation。

For now, THE performance of SQLX will be similar to that of Rbatis
so   You don't need to add Rbatis。

**Comment by suhanyujie** [comment]
> > 大佬在rust微信群里面吗？
> 
> ![908f7aabb643867cf75fc4e293e1203](https://user-images.githubusercontent.com/3481148/96582606-19e77080-130e-11eb-9584-092ca3aa3353.jpg)

二维码已过期。。。

**Comment by zhuxiujia** [comment]
> > > 大佬在rust微信群里面吗？
> > 
> > 
> > ![908f7aabb643867cf75fc4e293e1203](https://user-images.githubusercontent.com/3481148/96582606-19e77080-130e-11eb-9584-092ca3aa3353.jpg)
> 
> 二维码已过期。。。


先加微信 zxj347284221 拉进群
